test/SlowTests/Issues/RavenDB-23728.cs

```diff
@@ -0,0 +1,225 @@
+                }
+            }
+
+            for (int i = 0; i < 10; i++)
+            {
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new User { Id = "Users/1", Name = i.ToString() });
+                    await session.SaveChangesAsync();
+                }
+            }
+
+            for (int i = 0; i < 10; i++)
+            {
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(new User { Id = "Users/2", Name = GenerateLargeRandomString(32 * 1024) });
+                    await session.SaveChangesAsync();
+                }
+            }
+
+            configuration = new RevisionsConfiguration { Default = new RevisionsCollectionConfiguration { Disabled = false, MinimumRevisionsToKeep = 5 } };
+            await RevisionsHelper.SetupRevisions(store, Server.ServerStore, configuration: configuration);
+
+            // enforce
+            await WaitWithTimeoutAsync(() => EnforceConfiguration(store, new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "Users", "Companies" }), timeout: TimeSpan.FromSeconds(15));
+
+            await AssertRevisionsCountAsync(store, "Companies/1", 5);
+            await AssertRevisionsCountAsync(store, "Users/2", 5);
+
+            await AssertRevisionsCountAsync(store, "Users/1", 5); // Fail
+
+        }
+
+        public static string GenerateLargeRandomString(int sizeInBytes)
+        {
+            int length = sizeInBytes / 2; // Each char is 2 bytes
+            Random random = new Random();
+            StringBuilder stringBuilder = new StringBuilder(length);
+
+            for (int i = 0; i < length; i++)
+            {
+                // Generate a random lowercase letter from 'a' to 'z'
+                char randomChar = (char)random.Next('a', 'z' + 1);
+                stringBuilder.Append(randomChar);
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        private static async Task AssertRevisionsCountAsync(DocumentStore store, string id, int expectedCount)
+        {
+            using (var session = store.OpenAsyncSession())
+            {
+                var count = await session.Advanced.Revisions.GetCountForAsync(id);
+                Assert.Equal(expectedCount, count);
+            }
+        }
+
+        private async Task EnforceConfiguration(DocumentStore store, HashSet<string> collections = null)
+        {
+            var db = await Databases.GetDocumentDatabaseInstanceFor(store);
+            using (var token = new OperationCancelToken(db.Configuration.Databases.OperationTimeout.AsTimeSpan, db.DatabaseShutdown, CancellationToken.None))
+                await db.DocumentsStorage.RevisionsStorage.EnforceConfigurationAsync(_ => { }, includeForceCreated: false, collections, token: token);
+        }
+
+        private static async Task WaitWithTimeoutAsync(Func<Task> act, TimeSpan timeout)
```

Comment:
can we use the `token` that is passed to `EnforceConfigurationAsync` instaed?

Reply:
Fixed